app/controllers/organizations/staff/manage_fosters_controller.rb

```diff
@@ -1,8 +1,89 @@
 class Organizations::Staff::ManageFostersController < Organizations::BaseController
+  include ::Pagy::Backend
+
   layout "dashboard"
 
+  before_action :context_authorize, only: %i[new create index]
+  before_action :set_foster, only: %i[edit update destroy]
+
+  def new
+    @pets = Pet.fosterable.order(:name)
+    @accounts = AdopterFosterAccount.fosterers.order(:last_name)
+    @foster = Match.new
+  end
+
+  def create
+    @foster = Match.new(match_params.merge(match_type: :foster))
+
+    if @foster.save
+      redirect_to action: :index
+    else
+      @pets = Pet.fosterable.order(:name)
+      @accounts = AdopterFosterAccount.fosterers.order(:last_name)
+
+      render :new, status: :unprocessable_entity
+    end
+  end
+
   def index
-    authorize! User, context: {organization: Current.organization},
-      with: Organizations::ManageFostersPolicy
+    @q = authorized_scope(Match.fosters).joins(:pet).ransack(params[:q])
+    @pagy, paginated_fosters = pagy(
+      @q.result
+        .includes(:pet, :user)
+        .order("pets.updated_at DESC"),
+      items: 10
+    )
+    @foster_pets = paginated_fosters.group_by(&:pet)
```

Comment:
Using pagy and ransack together was painless but getting the sorting to work as expected with the grouping required some finessing. If anyone wants to update this in the future, be careful and read the docs.